vllm/model_executor/guided_decoding/__init__.py

```diff
@@ -40,43 +40,50 @@ def maybe_backend_fallback(
             guided_params.backend = "outlines"
 
     if guided_params.backend == "xgrammar":
-        # xgrammar only has x86 wheels for linux, fallback to outlines
-        from vllm.platforms import current_platform
-        if current_platform.get_cpu_architecture() is not CpuArchEnum.X86:
-            logger.warning("xgrammar is only supported on x86 CPUs. "
+        from vllm.model_executor.guided_decoding.xgrammar_decoding import (  # noqa
+            xgr_installed)
+        if not xgr_installed:
+            logger.warning("xgrammar module cannot be imported successfully. "
                            "Falling back to use outlines instead.")
             guided_params.backend = "outlines"
+        else:
```

Comment:
I don't think you need to move everything under a new `else:` block here. You can make the `if current_platform` check an `elif` like the rest of the checks in the xgrammar section here. I think that will keep this section a bit more consistent.

Reply:
Great suggestion. Thanks! Updated.